wandb/sdk/data_types.py

```diff
@@ -911,6 +925,105 @@ def __init__(self, data_or_path: Union[str, "TextIO"], **kwargs: str) -> None:
         else:
             raise ValueError("Data must be file name or a file object")
 
+    @classmethod
+    def from_rdkit(
+        cls,
+        data_or_path: "RDKitDataType",
```

Comment:
I am a bit confused as to why we accept both a file path and an rdkit object here. It seems like the options should be:

* Molecule.__init__(FILE_PATH)
* Molecule.from_rdkit(RDKIT_OBJ)
* Molecule.from_smiles(SMILES_STR)

Maybe I am not aware of some particular details of the filetypes which are marked as RDKIT types? Why can't those go in the standard constructor?

Furthermore, i see that there is some duplication in file type between the two (for example, `mol2` filetypes are supported in both) - under what conditions would the user choose to use `from_rdkit` over the standard constructor if they have a mol2 file?

Reply:
I totally see your point, let me explain. 

We are using [`nglviewer`](http://nglviewer.org/nglview/latest/) to render 3D molecules in the UI, so the currently supported file types simply reflect what it knows how to render. We do not apply any transformations to the input files, store them as is, and feed them straight into `nglviewer`; trying to feed in certain `rdkit` file types will result in an error:
![image](https://user-images.githubusercontent.com/7557205/141508955-2becf428-9ba5-4343-ab3b-83b233c7888d.png)

I see two ways on how to enable rendering of the `rdkit`-specific file types:
1. Convert them into something that `nglviewer` understands. This is what I implemented - I convert stuff into the `pdb` format that `nglviewer` natively understands. Separating this procedure from the standard constructor seemed the right thing to do as there are a couple of "knobs" in the conversion process that I think the users should have access to (plus the fact that there's conversion going on in the first place). 
2. Use a different framework on the frontend, such as [`3DMol`](http://3dmol.csb.pitt.edu/doc/tutorial-embeddable.html), that can render those file types natively, without having to convert anything.

If you agree that 1) is the way to go, I can put the conversion logic into the standard constructor.

Regarding the overlap in the filetypes - I put in all `rdkit`-supported types in there in case users want to have access to those "knobs", such as the 3D structure optimization that `rdkit` can do. Happy to pull them out to avoid potential confusion.